waterbutler/providers/rushfiles/provider.py

```diff
@@ -88,7 +88,24 @@ async def revalidate_path(self,
                               base: WaterButlerPath,
                               name: str,
                               folder: bool=None) -> WaterButlerPath:
-        raise NotImplementedError # Or user super if appropriate
+        response = await self.make_request(
+            'GET',
+            self.build_url(str(self.share['id']), 'virtualfiles', base.identifier, 'children'),
+            expects=(200, 404,),
+            throws=exceptions.MetadataError,
+        )
+        if response.status == 404:
+            raise exceptions.NotFoundError(name)
+        res = await response.json()
+        child_id, index = self._search_inter_id(res, name)
+
+        if res['Data'][index]['IsFile'] == folder:
```

Comment:
If file/folder is not found, `index` will be `None` and indexing the array with it will fail. The check for child_id should come first.